scylla-cdc/src/stream_generations.rs

```diff
@@ -0,0 +1,540 @@
+use futures::stream::StreamExt;
+use scylla::batch::Consistency;
+use scylla::frame::response::result::Row;
+use scylla::frame::value::Timestamp;
+use scylla::query::Query;
+use scylla::{IntoTypedRows, Session};
+use std::error::Error;
+use std::sync::Arc;
+use uuid::Uuid;
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Ord, PartialOrd)]
+pub struct Generation(chrono::Duration);
```

Comment:
A generation consists of a "generation timestamp" - which is a point in time when the generation starts operating - and a set of streams (stream IDs). The streams are part of the generation. This is just the timestamp, so perhaps `GenerationTimestamp` would be a better name?

Also, since the generation timestamp *is* a point in time, representing it using a `Duration` is probably not a good idea. If you want to represent a point in time using a `Duration` you also need to specify the epoch (the time point is then denoted by the epoch + the duration). A better type would probably be `Instant`.

Reply:
Ad 2:
I am not sure about that. In `std::time::Instant`'s docs it is said that:
`Instants are opaque types that can only be compared to one another. There is no method to get “the number of seconds” from an instant. Instead, it only allows measuring the duration between two instants (or comparing two instants).`

I may be mistaken, but I think using `Duration` is better since Scylla Rust Driver uses it internally for representing `Time` and `Timestamp`. 
And... I am not sure how to implement this using `Instant`.